### PR TITLE
Fix external DLL loading on wheels

### PR DESCRIPTION
**Pull request by andfoy**
This PR registers the path of the main torchvision library via `AddDllDirectory` on Python < 3.7 and `os.add_dll_directory` on Python >= 3.8. The extensions are preloaded using `LoadLibraryExW` to circunvent the limitation imposed by `AddDllDirectory` without calling `SetDefaultDllDirectories`

### CODE REVIEW

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

I'm not very comfortable with the current solution.

Would there be a way to fix Windows without having to do this PATH manipulation at import time?

For reference, we identified in https://github.com/pytorch/vision/pull/2775 that the Windows pip package for torchvision are still not loading `image.dll` correctly, and this PR is a tentative fix for it.

File: torchvision/io/_video_opt.py [hunk 1]
```diff
@@ -15,6 +15,11 @@
 try:
     lib_dir = os.path.join(os.path.dirname(__file__), "..")
 
+    if os.name == 'nt':
+        # This is required to load the wheel external binaries
+        os.environ['PATH'] = os.pathsep.join(
+            [os.path.abspath(lib_dir)] + [os.environ['PATH']])
```
[the remark this answers is not in the document]
Comment (andfoy) [reply torchvision/io/_video_opt.py:21]: I'm aware of `AddDllDirectory`, but I tried it without success: https://github.com/pytorch/pytorch/blob/master/torch/__init__.py#L92

[the remark this answers is not in the document]
Comment (andfoy) [reply torchvision/io/_video_opt.py:21]: Maybe the failure is related to this:

imagen

**Comment by andfoy** [comment]
It seems that this solution is damaging also the binary imports of other libraries, such as Scipy

**Comment by fmassa** [comment]
@peterjc123 could you help review this PR?

For context, we identified in #2775 that the Windows pip package for torchvision are still not loading image.dll correctly, and this PR is a tentative fix for it.

**Review by peterjc123 — APPROVED** [review]
LGTM. BTW, it would be better to refactor the code to eliminate duplicates.

**Comment by peterjc123** [comment]
> Would there be a way to fix Windows without having to do this PATH manipulation at import time?

No, unlike unix platforms, Windows doesn't have an alternative for RPATH for native executables.

**Comment by fmassa** [comment]
Thanks @andfoy and @peterjc123 , I'm merging this to unblock and do some testings.

Also, one thing that I would like to point out is that there is a non-symmetric inclusion here, where we add the path to the DLLs in `extension.py` but not in `image` and `video`, which might add the implicit requirement that we need to import `extension.py` before the others. Not sure if this is a problem?

**Pull request merged**